### PR TITLE
[clang-tidy][test] Add more bugprone-string-constructor tests

### PR DESCRIPTION
**Pull request by maflcko**
Test:

* string literal concatenation is accepted
* escaped embedded null char is accepted
* taking over the implicit null char is rejected

### CODE REVIEW

**Review by PiotrZSL — APPROVED** [review]
LGTM, wait until CI passes before merging

**Review by HerrCai0907 — APPROVED** [review]

**Pull request merged**
